chain/chunks/src/test_utils.rs

```diff
@@ -0,0 +1,440 @@
+use crate::{SealsManager, NUM_PARTS_REQUESTED_IN_SEAL};
+
+use near_chain::types::{ApplyTransactionResult, RuntimeAdapter};
+use near_chain::Error;
+use near_crypto::vrf::{Proof, Value};
+use near_crypto::Signature;
+
+use near_pool::types::PoolIterator;
+use near_primitives::block::BlockHeader;
+use near_primitives::challenge::{ChallengesResult, SlashedValidator};
+use near_primitives::errors::InvalidTxError;
+use near_primitives::hash::CryptoHash;
+use near_primitives::receipt::Receipt;
+use near_primitives::sharding::{ChunkHash, ShardChunkHeader};
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::BlockHeight;
+use near_primitives::types::{
+    AccountId, ApprovalStake, EpochId, ShardId, StateRoot, StateRootNode, ValidatorStake,
+    ValidatorStats,
+};
+use near_primitives::views::{EpochValidatorInfo, QueryRequest, QueryResponse};
+
+use near_store::{PartialStorage, ShardTries, Store, StoreUpdate, Trie};
+use std::cmp::Ordering;
+use std::sync::{Arc, Mutex};
+
+pub struct MockRuntimeAdapter {
```

Comment:
What's the reason for having a separate `MockRuntimeAdapter` in addition to `KeyValueRuntime`?

Reply:
`KeyValueRuntime` seemed like overkill for what I wanted to do in the unit tests. All I needed was for ~3 functions to work, and I didn't even really care about their internal logic of those functions.

I dislike the amount of useless boilerplate it introduces (all the `unimplemented` functions), and I could see an argument for using `KeyValueRuntime` to make the tests more realistic. But overall I think this approach isolates the exact logic I want to test and the tests themselves are relatively clean and easy to read (even if they rely on the mess of code in the new `test_utils`).

I will use the `KeyValueRuntime` if you think that is a better approach though.

Reply:
I think it depends on the amount of effort required here. I didn't read it very closely. Do you introduce something here that is not available in `KeyValueRuntime`? If it can work as a drop-in replacement I don't see the need to have a separate `MockRuntimeAdapter`.

Reply:
The reason of using same KVRuntime is to make supporting easier. I'm okay having several Testing Runtimes for separate purposes if they cannot be interchanged.

I see, in this case it might be easier to handle `part_owner` and `num_parts` for testing. Honestly, I see only two usages of `MockRuntimeAdapter` and both are in sanity tests. I'm not convinced it's enough to introduce new Runtime.

Reply:
Switched to using `KeyValueRuntime` in https://github.com/nearprotocol/nearcore/pull/2702/commits/dcc9f796cebd158dca09b09e3947bc9d4ce002f9